### DISCUSSION TITLE
Setting functions on an object

### DISCUSSION THREAD

**Discussion by pgundlach** [dop251/goja#450] [Q&A] [answered]
I'd like to create an object (here a fictional “document”) and call methods on it. I can set a function like `newdoc` which creates a document, but I don't know how to wrap the returned object so that I can attach functions on it. Newbee warning, so I might look at the wrong place.

How do I implement the `d.dosomething()` which is commented out in the js source below?

```go
package main

import (
	"fmt"
	"os"

	"github.com/dop251/goja"
)

type document struct{}

func newDocument() *document {
	return &document{}
}

func (d *document) DoSomething() {
}

func main() {
	runtime := goja.New()

	gjNewDoc := func(call goja.FunctionCall) goja.Value {
		return runtime.ToValue(newDocument())
	}
	runtime.Set("newdoc", gjNewDoc)

	v, err := runtime.RunString(`
var d = newdoc();
// d.dosomething();
`)
	if err != nil {
		fmt.Println(err)
		os.Exit(-1)
	}
	fmt.Println(v)
}
```

**Comment by dop251** [comment, marked as answer]
Methods are already exposed, as long as they are exported, so in your example `d.DoSomething()` would work.

**Reply by pgundlach** [reply]
That is cool, I have not seen / anticipated that. Thank you!
